### PR TITLE
Fix various lint

### PR DESCRIPTION
**Pull request by hauntsaninja**

### CODE REVIEW

File: mypy/semanal.py [hunk 1]
```diff
@@ -4962,7 +4962,7 @@ def bind_name_expr(self, expr: NameExpr, sym: SymbolTableNode) -> None:
         """Bind name expression to a symbol table node."""
         if isinstance(sym.node, TypeVarExpr) and self.tvar_scope.get_binding(sym):
             self.fail(
-                '"{}" is a type variable and only valid in type ' "context".format(expr.name), expr
+                '"{}" is a type variable and only valid in type context'.format(expr.name), expr
```
Comment (AlexWaygood) [inline mypy/semanal.py:4965]:
```suggestion
                f'"{expr.name}" is a type variable and only valid in type context', expr
```

File: mypy/stubdoc.py [hunk 2]
```diff
@@ -325,7 +325,7 @@ def args_kwargs(signature: FunctionSig) -> bool:
             return has_arg("*args", signature) and has_arg("**kwargs", signature)
 
         # Move functions with (*args, **kwargs) in their signature to last place.
-        return list(sorted(self.signatures, key=lambda x: 1 if args_kwargs(x) else 0))
+        return sorted(self.signatures, key=lambda x: 1 if args_kwargs(x) else 0)
```
Comment (AlexWaygood) [inline mypy/stubdoc.py:328]:
```suggestion
        return sorted(self.signatures, key=lambda x: bool(args_kwargs(x)))
```

File: mypyc/codegen/emitmodule.py [hunk 3]
```diff
@@ -993,7 +993,7 @@ def _toposort_visit(name: str) -> None:
             result.append(decl.declaration)
             decl.mark = True
 
-        for name, marked_declaration in marked_declarations.items():
+        for name in marked_declarations.keys():
```
Comment (AlexWaygood) [inline mypyc/codegen/emitmodule.py:996]: Or just iterate directly over the dict:

```suggestion
        for name in marked_declarations:
```

File: mypy/stubdoc.py (hunk 2 above)
Comment (hauntsaninja) [reply mypy/stubdoc.py:328]: I left this one alone because it's not the most obvious what the sort order of booleans is

Comment (AlexWaygood) [reply mypy/stubdoc.py:328]: Fair

File: pyproject.toml [hunk 4]
```diff
@@ -33,24 +33,26 @@ fix = true
 select = [
   "E",       # pycodestyle (error)
   "F",       # pyflakes
+  "W",       # pycodestyle (warning)
   "B",       # flake8-bugbear
   "I",       # isort
   "RUF100",  # Unused noqa comments
   "PGH004",  # blanket noqa comments
   "UP",      # pyupgrade
+  "C4", "SIM201", "ISC001", "RET501",
```
Comment (AlexWaygood) [inline pyproject.toml:42]: I quite like having comments for each rule here, like we have for the existing items in this list – but don't feel strongly about it :)

**Review by AlexWaygood — APPROVED** [review]

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Pull request merged**